src/main/java/com/iexec/sms/encryption/EncryptionService.java

```diff
@@ -120,4 +130,23 @@ boolean checkOrFixReadOnlyPermissions(String aesKeyPath) {
         }
         return true;
     }
+
+    /**
+     * Enables the file to be written
+     * Use for restoration process
+     *
+     * @return {@code true} if the switch was successful, {@code false} if any error occurs.
+     */
+    public boolean setWritePermissions() {
```

Comment:
Wouldn't we prefer to return `File.canWrite()` in this method ?
What is more important to us in this specific situation, the goal or the journey ?

Reply:
I mean this would return `false` if the file was already writable

Reply:
Or should we add a protection as done in `checkOrFixReadOnlyPermissions` ?